src/PrestaShopBundle/ApiPlatform/Processor/CommandProcessor.php

```diff
@@ -31,32 +31,44 @@
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProcessorInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
+use PrestaShopBundle\ApiPlatform\DomainSerializer;
+use ReflectionException;
+use Symfony\Component\Serializer\Exception\ExceptionInterface;
 
 class CommandProcessor implements ProcessorInterface
 {
-    /**
-     * @var CommandBusInterface
-     */
-    private $commandBus;
-
     /**
      * @param CommandBusInterface $commandBus
+     * @param DomainSerializer $apiPlatformSerializer
      */
-    public function __construct(CommandBusInterface $commandBus)
-    {
-        $this->commandBus = $commandBus;
+    public function __construct(
+        private readonly CommandBusInterface $commandBus,
+        private readonly DomainSerializer    $apiPlatformSerializer,
+    ) {
     }
 
+    /**
+     * @param $data
+     * @param Operation $operation
+     * @param array $uriVariables
+     * @param array $context
+     *
+     * @return void
+     *
+     * @throws NoExtraPropertiesFoundException
+     * @throws ExceptionInterface|ReflectionException
+     */
     public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $command = $operation->getExtraProperties()['command'];
-        $reflectionMethod = new \ReflectionMethod($command, '__construct');
-        $constructParameters = $reflectionMethod->getParameters();
-        $parameters = [];
-        foreach ($constructParameters as $parameter) {
-            $parameters[] = $data->{$parameter->name};
+        $commandClass = $operation->getExtraProperties()['command'] ?? null;
+
+        if (null === $commandClass) {
```

Comment:
```suggestion
        if (null === $commandClass || !class_exists($commandClass)) {
```

Suggestion for later we could check if the provided class exist